### ISSUE TITLE
add support for ignoring method with annotation

### ISSUE THREAD

**Issue by monperrus** [STAMP-project/pitest-descartes#134]
following-up on @Kvarnefalk's thesis
 
we need to give a way for developers to exclude a method (as `// NOSONAR` in Sonarqube)

since we're operating on bytecode and at the method level, we can do this by adding an annotation to method:

```
@DescartesIgnore
void pseudoTested() {
  // we ignore this method in Descartes because of the annotation
  ...
}
```
 
WDYT?

**Comment by oscarlvp** [comment]
This should be easy to implement. However, it would add a dependency to the project just to add this annotation. 
More relevant: PIT can be configured to avoid mutating specific methods. Isn't this enough? Is there any particular case that is  not be covered by this configuration? Check the `excludedMethods` in  http://pitest.org/quickstart/maven/

**Comment by monperrus** [comment]
> However, it would add a dependency to the project just to add this annotation.

yes, note that there are many annotation only libraries on Maven Central.

> More relevant: PIT can be configured to avoid mutating specific methods. I

Functionally, it's equivalent but it's not the same user experience. It would mean changing another file. The annotation enables to annotate the file already modified in the PR, on where Descartes was run specifically in CI. The process seems smoother.

**Comment by oscarlvp** [comment]
Included in 1.3
